src/main/scala/analysis/Local.scala

```diff
@@ -0,0 +1,351 @@
+//            assert(graph.pointTo.contains(node.getCell(offset))) TODO
+            result = graph.find(graph.find(node.getCell(offset)).getPointee._1)
+          else
+            graph.selfCollapse(node)
+      )
+    val resultOffset = result.offset
+    graph.selfCollapse(result.node.get)
+    result.node.get.getCell(result.offset)
+
+  /**
+   * handles unsupported pointer arithmetic by collapsing all the nodes invloved
+   */
+  def unsupportedPointerArithmeticOperation(n: CFGPosition, expr: Expr, lhsCell: DSC): Unit = {
+    var containsPointer = false
+    breakable {
+      for (v <- expr.variables) {
+        if varToSym.contains(n) && varToSym(n).contains(v) then
+          containsPointer = true
+          break
+      }
+    }
+    if containsPointer then
+      val cell = expr.variables.foldLeft(lhsCell) {
+        (c, v) =>
+          val cells: Set[Slice] = graph.getCells(n, v)
+
+          cells.foldLeft(c) {
+            (c, p) =>
+              graph.mergeCells(c, p._1)
+          }
+      }
+      val node = cell.node.get
+      node.flags.unknown = true
+      val test = graph.collapseNode(node)
+  }
+
+  def visit(n: CFGPosition): Unit = {
+    if visited.contains(n) then
+      return
+    else
+      visited.add(n)
+    n match
+      case DirectCall(proc, target, label) if proc.name == "malloc" => // R0 = Malloc()
+        val size: BigInt = evaluateExpression(mallocRegister, constProp(n)) match
+          case Some(value) => value.value
+          case None => 0
+        val node = DSN(Some(graph), size)
+        node.allocationRegions.add(HeapLocation(nextMallocCount, proc, size))
+        node.flags.heap = true
+        graph.mergeCells(graph.varToCell(n)(mallocRegister)._1, node.cells(0))
+      case call: DirectCall if params.contains(call.target) => // Rx, Ry, ... Rn = FunctionCall()
+        // create call sites for the callees
+        val cs = CallSite(call, graph)
+        graph.callsites.add(cs)
+        cs.paramCells.foreach{
+          case (variable: Variable, slice: Slice) =>
+            visitPointerArithmeticOperation(call, graph.adjust(slice), variable, 0)
+        }
+        cs.returnCells.foreach{
+          case (variable: Variable, slice: Slice) =>
+            val returnArgument  = graph.varToCell(n)(variable)
+            graph.mergeCells(graph.adjust(returnArgument), graph.adjust(slice))
+        }
+      case Assign(variable: Variable, rhs: Expr, maybeString) =>
+        val expr: Expr = unwrapPaddingAndSlicing(rhs)
+        val lhsCell = graph.adjust(graph.varToCell(n)(variable))
+        var global = isGlobal(rhs, n)
+        var stack = isStack(rhs, n)
+        if global.isDefined then // Rx = global address
+          graph.mergeCells(lhsCell, global.get)
+        else if stack.isDefined then // Rx = stack address
+          graph.mergeCells(lhsCell, stack.get)
+        else
+          expr match
+            case BinaryExpr(op, arg1: Variable, arg2) => // Rx = Rx + c
```

Comment:
Why does this treat Rx = Rx + c differently to Rx = c + Rx (which is not handled)?

Reply:
do you know of an example of Rc = c + Rx?

Reply:
It's better practice to handle all inputs that we can, rather than only matching a very narrow set of inputs that we're familiar with. Making assumptions like this can cause problems later on - the less assumptions about what the input looks like, the better.